flagscale/runner/runner_serve.py

```diff
@@ -22,6 +24,141 @@
 )
 
 
+def _get_multiple_free_ports(num=1, exclude_ports=[]):
+    allocated_ports = []
+    for i in range(num):
+        port = get_free_port()
+        while port in allocated_ports or port in exclude_ports:
+            port = get_free_port()
+        allocated_ports.append(port)
+    return allocated_ports
+
+
+class ResourceManager:
```

Comment:
ResourceManager can be utilized not only for serving but also for other components. Therefore, it can be extracted for future reuse.

Reply:
Nice idea. Done